ci: add linting to project

romanrostislavovich opened the pull request:
added next gulp commands:
- `lint-js`  - check all js files in folder `src` via `eslint`
- `lint-json` - check all json files in folder `src` via `jsonFormat`

added next gulp parallel commands:
- `lint` - include: `lint-js` and `lint-json`

Closes #15

xferra commented on .eslintrc.json:
```diff
@@ -3,5 +3,9 @@
+    "rules":{
+        "linebreak-style": 0,
+        "max-len": [2, {"code": 120, "tabWidth": 4, "ignoreUrls": true}]
+    }
```
why those rules were added?

svoboda-rabstvo replied:
linebreak-style - detected `\r` in `linux` or `\r\n` in `windows`.  
max-len:
- `code` max len for one line, because some gulp.tasks so long (by default this value is 80)

xferra replied:
1) should be solved via git config
2) rewrite them ;)

svoboda-rabstvo replied:
done

xferra commented on package.json:
```diff
@@ -25,15 +25,15 @@
+    "gulp-eslint": "4.0.2",
+    "gulp-hub": "0.8.0",
+    "gulp-json-format": "2.0.0",
+    "gulp-json-schema": "1.0.0",
```
Why are some versions pinned ant the rest no?

svoboda-rabstvo replied:
done

xferra commented on script/gulp/index.js:
```diff
@@ -19,7 +20,7 @@ const amd = {
 
 // Disable logging
-// amd.jsonSchemaRemote.setLoggingFunction(() => { });
+amd.jsonSchemaRemote.setLoggingFunction(() => { });
```
?

svoboda-rabstvo replied:
It's for `json-schema-remote`, if it's disable, then log will be added some logs when schemas  download by url

xferra replied:
I think they at least useful

svoboda-rabstvo replied:
done

xferra commented on script/gulp/task.test.js:
```diff
@@ -25,18 +25,22 @@ const testTrue = (test, schema, done) => {
         log.error(`FILE: ${test.data.$ref}`);
-        log.error(`MESSAGE: ${error.errors[0].message}`);
-        log.error(`SCHEMA PATH: ${error.errors[0].schemaPath}`);
+        if (error.errors != null) {
```
Please use `===` or just `if (error.errors)`

romanrostislavovich committed:
ci: add linting to project

added next gulp commands:
- `lint-js`  - check all js files in folder `src` via `eslint`
- `lint-json` - check all json files in folder `src` via `eslint`

added next gulp parallel commands:
- `lint` - include: `lint-js` and `lint-json`

Closes #15

romanrostislavovich committed:
Merge branch 'develop' into develop

xferra reviewed (approved):
LGTM
